src/modm/driver/rtc/mcp7941x.hpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2021, Odin Holmes
+ * Copyright (c) 2021, Raphael Lehmann
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * chip datasheet available at http://ww1.microchip.com/downloads/en/DeviceDoc/22266A.pdf
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_MCP7941X_HPP
+#define MODM_MCP7941X_HPP
+
+#include <array>
+
+#include <modm/architecture/interface/register.hpp>
+#include <modm/architecture/interface/i2c_device.hpp>
+#include <modm/processing/protothread.hpp>
+
+namespace modm
+{
+
+// forward declaration for friending with mcp7941::Data
+template < class I2cMaster >
+class Mcp7941x;
+
+/// @ingroup modm_driver_mcp7941
+struct mcp7941x
+{
+
+public:
+
+	//see datasheet section 4 for more information on format
+	struct
+	DateTime{
+		uint8_t days;
+		uint8_t months;
+		uint8_t years;
+		uint8_t seconds;
+		uint8_t minutes;
+		uint8_t hours;
+	};
+
+	struct modm_packed
+	Data
+	{
+		template < class I2cMaster >
+		friend class Mcp7941x;
+
+	public:
+		constexpr DateTime getDateTime(){
+			return date_time;
+		}
+		uint8_t unique_id[8];
+	private:
+		DateTime date_time;
+	};
+	uint8_t scratch[8];
+
+
+protected:
+	constexpr inline uint8_t
```

Comment:
`constexpr` implies inline. From the standard, section [dc.constexpr]:

>  `A function or static data member declared with the constexpr or consteval specifier is implicitly an inline function or variable ([dcl.inline])`

```suggestion
	constexpr uint8_t
```